frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/spdx/SpdxPortlet.java

```diff
@@ -0,0 +1,415 @@
+
+        if (addNew) {
+            spdxDocumentData = (String) request.getAttribute(SPDXDocument._Fields.TYPE.toString());
+            documentCreationInfoData = (String) request.getAttribute(SPDXDocument._Fields.SPDX_DOCUMENT_CREATION_INFO_ID.toString());
+            packageInfoData = (String) request.getAttribute(SPDXDocument._Fields.SPDX_PACKAGE_INFO_IDS.toString());
+        } else {
+            spdxDocumentData = request.getParameter(SPDXDocument._Fields.TYPE.toString());
+            documentCreationInfoData = request.getParameter(SPDXDocument._Fields.SPDX_DOCUMENT_CREATION_INFO_ID.toString());
+            packageInfoData = request.getParameter(SPDXDocument._Fields.SPDX_PACKAGE_INFO_IDS.toString());
+        }
+        if (!isNullOrEmpty(spdxDocumentData)) {
+            SPDXDocument spdx = parseSPDXDocumentFromRequest(spdxDocumentData);
+            SPDXDocumentService.Iface spdxClient = new ThriftClients().makeSPDXClient();
+            if (spdx != null) {
+                if (isNullOrEmpty(spdx.getReleaseId()) && !isNullOrEmpty(releaseId)) {
+                    spdx.setReleaseId(releaseId);
+                }
+                if (isNullOrEmpty(spdx.getId())) {
+                    spdx.unsetId();
+                    spdx.unsetRevision();
+                    spdxDocumentId = spdxClient.addSPDXDocument(spdx, user).getId();
+                } else {
+                    spdxClient.updateSPDXDocument(spdx, user);
+                    spdxDocumentId = spdx.getId();
+                }
+            }
+        }
+        if (!isNullOrEmpty(documentCreationInfoData)) {
+            DocumentCreationInformation document = parseDocumentCreationInfoFromRequest(documentCreationInfoData);
+            if (document != null) {
+                DocumentCreationInformationService.Iface documentClient = new ThriftClients().makeSPDXDocumentInfoClient();
+                if (isNullOrEmpty(document.getSpdxDocumentId())) {
+                    document.setSpdxDocumentId(spdxDocumentId);
+                }
+                if (isNullOrEmpty(document.getId())) {
+                    document.unsetId();
+                    document.unsetRevision();
+                    documentClient.addDocumentCreationInformation(document, user);
+                } else {
+                    documentClient.updateDocumentCreationInformation(document, user);
+                }
+            }
+        }
+        if (!isNullOrEmpty(packageInfoData)) {
+            Set<PackageInformation> packageInfos = parsePackageInfosFromRequest(packageInfoData);
+            SPDXDocumentService.Iface SPDXClient = new ThriftClients().makeSPDXClient();
+            SPDXDocument spdxDocument = SPDXClient.getSPDXDocumentById(spdxDocumentId, user);
+            idDeletePackageInfo(packageInfos, spdxDocument, user);
+            if (packageInfos != null) {
+                PackageInformationService.Iface packageClient = new ThriftClients().makeSPDXPackageInfoClient();
```

Comment:
Kindly cerate a `ThriftClient` object on top of method body.
`final ThriftClients thriftClients = new ThriftClients();` 
and reuse it every other place `thriftClients.makeSPDXClient();`

Reply:
I fixed it. Thank you for your review.